src/components/Editor/SearchBar.js

```diff
@@ -156,7 +156,8 @@ class SearchBar extends Component<Props, State> {
 
     if (this.props.searchOn && editor) {
       const selection = editor.codeMirror.getSelection();
-      this.setState({ query: selection });
+      let query = selection || this.$input.$input.value || "";
```

Comment:
This likely needs some work but you get the idea.

Reply:
Interesting! I see how you are solving this, and i agree: its inverting the flow of the two components and directly reaching into the subcomponent, and that doesn't feel right. 

not 100% sure this will work, but have you tried this?:

```javascript
const query = selection ||this.state.query || "";
this.setState({ query: "" }); // clear it to reset the searchInput;
this.setState({ query });  // reset it as needed
```